scipy/stats/_axis_nan_policy.py

```diff
@@ -344,13 +356,30 @@ def axis_nan_policy_wrapper(*args, _no_deco=False, **kwds):
 
             # rename avoids UnboundLocalError
             if callable(n_samples):
+                # Future refactoring idea: no need for callable n_samples.
+                # Just replace `n_samples` and `kwd_samples` with a single
+                # list of the names of all samples, and treat all of them
+                # as `kwd_samples` are treated below.
                 n_samp = n_samples(kwds)
             else:
                 n_samp = n_samples or len(args)
 
+            # If necessary, rearrange function signature: accept other samples
+            # as positional args right after the first n_samp args
+            kwd_samp = [name for name in kwd_samples
+                          if kwds.get(name, None) is not None]
```

Comment:
scipy/stats/_axis_nan_policy.py:370:27: E127 continuation line over-indented for visual indent
```suggestion
                        if kwds.get(name, None) is not None]
```